scalableminds/webknossos
# Upgrade ThreeJS and use WebGL 2

### Summary
- Enable WebGL2
- Update ThreeJS to r137 to fix issues with RG textures
- Replace GLSL validation package to support webgl 2
- Update node to 16 since new glsl package requires this
- Remove shader rendering spec due to incompatibilities with node 16 (see https://app.circleci.com/pipelines/github/scalableminds/webknossos/7286/workflows/e17826f1-461a-481c-ba60-246174b32df5/jobs/18622/parallel-runs/0/steps/0-109). there was only one spec, anyway, with very low impact. the e2e screenshot tests serve us better here.


The screenshots have slightly thicker borders (the red, green, blue ones) now apparently:
old: 
![image](https://user-images.githubusercontent.com/2486553/181024307-68521305-8a25-416b-9c0f-2e31df6d0cef.png)

new: 
![image](https://user-images.githubusercontent.com/2486553/181024277-fa478104-f9dd-4899-8981-778eab8bc70a.png)

I'm not sure whether this is only in the screenshot tests or for the normal browser setup, too. However, I don't think that this is a blocker. Especially, because the pixel difference is so small that the screenshot tests pass.

### URL of deployed dev instance (used for testing):
- https://webgl.webknossos.xyz

### Todo:
- [x] clarify what cross sections are and check whether they work? (see cube.ts)

### Steps to test:
- [X] test on iPad (I already did this via browserstack and the rendering seems to work)
- [X] run nightly screenshot test against this branch
Also, try to test wk rather broad regarding rendering, but the following were changed during this PR which is why they should be tested with extra care:
- crosshair
	- in ortho
	- in flight mode
- mesh smoothing / computeVertexNormals
- Quaternion usage in 3D viewport / use the rotate buttons
- rendering of "cubes"

### Issues:
- fixes #5745 

------
(Please delete unneeded items, merge only when none are left open)
